### ISSUE TITLE
Notarization Timeout?

### ISSUE THREAD

**Issue by mikeKane** [fastlane/fastlane#16156]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject
Is there a way to Timeout an action? 

#### Question Description
Yesterday I was sending a Mac app for notarization, usually it takes 2-5 minutes but this was stuck for over 90 minutes. Is there a way to add a timeout for any action?

**Comment by mikeKane** [comment]
Not sure this is needed

**Comment by Cyberbeni** [comment]
Yeah, we have like a 2 hour timeout for it in our version that was once reached due to outage at Apple.

There is a sleep(30) in this loop, you can also add a counter to it and make it an option with basically infinite timeout to keep the current behaviour: https://github.com/fastlane/fastlane/blob/ad1cf01e24191db83d45d1187b0ce3e1c025fe75/fastlane/lib/fastlane/actions/notarize.rb#L55

**Comment by Cyberbeni** [comment]
But it would just ensure that in case you have iOS apps in the queue, they won't get blocked by notarization service outage. If you only want to build mac apps then it wouldn't really help, you would still have to wait for this service to come back online.

**Comment by mikeKane** [comment]
Thank you, but I would like to timeout the process after 30 minutes. Is this possible without modifying the code? If not I guess it would be nice to have that as an option in that action.

**Comment by mikeKane** [comment]
ive modified my code 

```
    timer = 0
        while notarization_info.empty? || (notarization_info['Status'] == 'in progress')
            
          if timer > 1800
               UI.message('Waiting to query request status')
               break
          end
```
seems to work fine.
